pkg/spec/operation.go

```diff
@@ -312,24 +319,52 @@ func CloneOperation(op *spec.Operation) (*spec.Operation, error) {
 	return &out, nil
 }
 
+func handleAuthJWT(operation *spec.Operation, bearerToken string) (*spec.Operation, bool) {
+	// Parse the claims without validating (since we don't want to bother downloading a key)
+	parser := jwt.Parser{}
+	token, _, err := parser.ParseUnverified(bearerToken, jwt.MapClaims{})
+	if err != nil {
+		return operation, false
+	}
+	claims, ok := token.Claims.(jwt.MapClaims)
+	if !ok {
+		return operation, false
+	}
+	if scope, ok := claims["scope"]; ok {
+		scopes := strings.Split(scope.(string), " ")
+		log.Debugf("Found scopes: %v", scopes)
+		return addSecurity(operation, OAuth2SecurityDefinitionKey, scopes...), true
+	}
+	log.Warnf("No scope defined in this token")
+	return addSecurity(operation, OAuth2SecurityDefinitionKey, []string{}...), true
+}
+
 func handleAuthReqHeader(operation *spec.Operation, sd spec.SecurityDefinitions, value string) (*spec.Operation, spec.SecurityDefinitions) {
 	if strings.HasPrefix(value, BasicAuthPrefix) {
 		operation = addSecurity(operation, BasicAuthSecurityDefinitionKey)
 		sd = updateSecurityDefinitions(sd, BasicAuthSecurityDefinitionKey)
 	} else if strings.HasPrefix(value, BearerAuthPrefix) {
-		operation = addSecurity(operation, OAuth2SecurityDefinitionKey)
-		sd = updateSecurityDefinitions(sd, OAuth2SecurityDefinitionKey)
+		tokenString := strings.TrimPrefix(value, BearerAuthPrefix)
+		if len(tokenString) > 0 {
+			// Note: it might not be a JWT so we don't update in that case.
+			operation, _ = handleAuthJWT(operation, tokenString)
```

Comment:
why not fallback to old behavior? 
```
			if !ok {
				operation = addSecurity(operation, OAuth2SecurityDefinitionKey)
			}
```

Reply:
Would it be better then to move lines 338-339 to 351 as the fallback behavior to make it more clear?

Reply:
Like other calls to `handleAuthJWT` you had a fallback in case any error occurred during `handleAuthJWT` run so I assume it should be done here.